climada/util/api_client.py

```diff
@@ -486,3 +533,166 @@ def purge_cache(self, local_path):
+        hazard_concat.sanitize_event_ids()
+        hazard.check()
+        return hazard_concat
+
+    def get_exposures(self, exposures_type, dump_dir=SYSTEM_DIR, **kwargs):
+        """Queries the data api for exposures datasets of the given type, downloads associated
+        hdf5 files and turns them into a climada.entity.exposures.Exposures object.
+
+        Parameters
+        ----------
+        hazard_type : str
+            Type of climada exposures.
+        dump_dir : str, optional
+            Directory where the files should be downoladed. Default: SYSTEM_DIR
+            If the directory is the SYSTEM_DIR, the eventual target directory is organized into
+            dump_dir > hazard_type > dataset name > version
+        **kwargs :
+            additional parameters passed on to `Client.get_datasets`
+
+        Returns
+        -------
+        climada.entity.exposures.Exposures
+            The combined exposures object
+        """
+        if 'data_type' in kwargs:
+            raise ValueError("data_type is already given as hazard_type")
+        if not exposures_type in EXP_TYPES:
+            raise ValueError("Valid exposures types are a subset of CLIMADA exposures types."
+                             f" Currently these types are supported: {EXP_TYPES}")
+        datasets = self.get_datasets(data_type=exposures_type, **kwargs)
+        if not datasets:
+            raise ValueError("no datasets found meeting the requirements")
+        return self.to_exposures(datasets, dump_dir)
+
+    def to_exposures(self, datasets, dump_dir=SYSTEM_DIR):
+        """Downloads hdf5 files belonging to the given datasets reads them into Exposures and
+        concatenates them into a single climada.Exposures object.
+
+        Parameters
+        ----------
+        datasets : list of DatasetInfo
+            Datasets to download and read into climada.Exposures objects.
+        dump_dir : str, optional
+            Directory where the files should be downoladed. Default: SYSTEM_DIR
+            If the directory is the SYSTEM_DIR, the eventual target directory is organized into
+            dump_dir > exposures_type > dataset name > version
+
+        Returns
+        -------
+        climada.entity.exposures.Exposures
+            The combined exposures object
+        """
+        exposures_list = []
+        for dataset in datasets:
+            target_dir = self._organize_path(dataset, dump_dir) \
+                         if dump_dir == SYSTEM_DIR else dump_dir
+            for dsf in dataset.files:
+                if dsf.file_format == 'hdf5':
+                    exposures_file = self.download_file(target_dir, dsf)
+                    exposures = Exposures()
+                    exposures.read_hdf5(exposures_file)
+                    exposures_list.append(exposures)
+        if not exposures_list:
+            raise ValueError("no exposures files found in datasets")
+
+        exposures_concat = Exposures()
```

Comment:
Here I only think it makes sense to concat country datasets but not different litpop datasets with different parameters